src/WidgetBase.ts

```diff
@@ -241,6 +248,33 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		this._checkOnElementUsage();
 	}
 
+	protected meta<T extends MetaBase>(MetaType: WidgetMetaConstructor<T>): T {
+		let cached = this._metaMap.get(MetaType);
+		if (!cached) {
+			const boundInvalidate = this.invalidate.bind(this);
+			const invalidate = function () {
+				global.cancelAnimationFrame(scheduled);
+				scheduled = global.requestAnimationFrame(boundInvalidate);
+			};
+			let scheduled: number | undefined;
+			cached = new MetaType({
+				nodes: this._nodeMap,
+				invalidate,
+				requireNode: (key: string) => {
```

Comment:
this builds up the required nodes, and can actually be an implementation detail i think?